### ISSUE TITLE
[BUG] npm list --json produces invalid json when --silent is included

### ISSUE THREAD

**Issue by csvan** [npm/cli#2163]
### Current Behavior:
In a situation where an error occurs but is suppressed by the `--silent` flag when running `npm list --json`, the resulting json output is invalid.

For example, the resulting json can look like this:

```
{
  "version": "0.0.0",
  "name": "@myscope/mypackage",
  "problems": [],
  "dependencies": {}
}
{
  "error": {
    "code": "ELSPROBLEMS",
    "summary": "invalid: @pm2/pm2-version-check@1.0.3 /pathtoproject/node_modules/@pm2/pm2-version-check\nmissing: react-dom@>= 16.8.0, required by styled-components@5.2.1\nmissing: react@>= 16.8.0, required by styled-components@5.2.1\nmissing: ts-node@>=9.0.0, required by jest-config@26.6.3\nmissing: bufferutil@^4.0.1, required by ws@7.2.5\nmissing: utf-8-validate@^5.0.2, required by ws@7.2.5\nmissing: bufferutil@^4.0.1, required by ws@7.3.1\nmissing: utf-8-validate@^5.0.2, required by ws@7.3.1\nmissing: canvas@^2.5.0, required by jsdom@16.4.0",
    "detail": ""
  }
}

```

These are two json objects in succession without a valid separator, trying to run e.g. `JSON.parse()` on it will fail since it is invalid.

### Expected Behavior:
The resulting output should only contain the result of `npm list --json`, and that result should be valid json.

### Environment:
- OS: OSX Catalina
- Node: 15.0.0
- npm: 7.0.0

Note that this most likely applies to npm 6 as well.

**Comment by ljharb** [comment]
What happens if you, in turn, redirect stdout and then stderr to `/dev/null`?

**Comment by csvan** [comment]
This seems to be the same issue as https://github.com/npm/cli/issues/2150, which was closed just now. Will see if I can still reproduce after that.

**Comment by csvan** [comment]
Still an issue on npm 7.0.11

**Comment by csvan** [comment]
Fixed in npm 7.0.11, where the `err` content will now be printed to console whereas `stdout` will correctly end up in the piped file.
